virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const (
+	OVA     = "ova"
+	vSphere = "vSphere"
+	DIR     = "/var/tmp/v2v"
+	FS      = "/mnt/disks/disk[0-9]*"
+	Block   = "/dev/block[0-9]*"
+	VDDK    = "/opt/vmware-vix-disklib-distrib"
+)
+
+var (
+	xmlFilePath string
+	server      *http.Server
+)
+
+func main() {
+	virtV2vArgs := []string{"virt-v2v", "-v", "-x"}
+	source := os.Getenv("V2V_source")
```

Comment:
I miss a check of whether `source` is either ova or vSphere but I don't see it in the entrypoint script either...

Reply:
I think we can assume its OK, since its being sent and verified from the controller side.

Reply:
i think that without it we will just fail quickly..
we do have the assumption that the controller will provide it, but since we also have this assumption on other variables and we do validate them - we can also validate this